EasyPost/Utilities/Rates.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace EasyPost.Utilities
+{
+    public static class Rates
+    {
+        /// <summary>
+        ///     Get the lowest rate from a list of rates
+        /// </summary>
+        /// <param name="rates">List of rates to parse.</param>
+        /// <param name="includeCarriers">Carriers to include in the filter.</param>
+        /// <param name="includeServices">Services to include in the filter.</param>
+        /// <param name="excludeCarriers">Carriers to exclude in the filter.</param>
+        /// <param name="excludeServices">Services to exclude in the filter.</param>
+        /// <returns>Lowest rate matching the filter.</returns>
+        public static Rate GetLowestObjectRate(List<Rate> rates, List<string>? includeCarriers = null, List<string>? includeServices = null, List<string>? excludeCarriers = null, List<string>? excludeServices = null)
```

Comment:
This looks a bit more verbose than the original, what changed between the two? As this is a highly trafficked function, we'd like to keep the behavior the same as before but just make it into a utility.